File: src/main/kotlin/com/contaduria/movimientofinanciero/repositories/AdministrativeDocumentRepository.java
package com.contaduria.movimientofinanciero.repositories;

import com.contaduria.movimientofinanciero.entities.AdministrativeDocument;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;

import java.util.Optional;

public interface AdministrativeDocumentRepository extends JpaRepository<AdministrativeDocument, Long>, JpaSpecificationExecutor<AdministrativeDocument> {

    Page<AdministrativeDocument> findAll(Specification<AdministrativeDocument> spec, Pageable pageable);
    public Optional<AdministrativeDocument> findByYearAndCodeOrganismAndNumber(Integer year, Integer codeOrganism, Integer number);

}

File: src/main/kotlin/com/contaduria/movimientofinanciero/repositories/AdministrativeDocumentRepository.kt
package com.contaduria.movimientofinanciero.repositories

import com.contaduria.movimientofinanciero.entities.AdministrativeDocument
import org.springframework.data.repository.PagingAndSortingRepository
import org.springframework.data.rest.core.annotation.RepositoryRestResource
import org.springframework.web.bind.annotation.CrossOrigin


@CrossOrigin(origins = ["*"])
@RepositoryRestResource(path = "administrative-documents")
interface AdministrativeDocumentRepository : PagingAndSortingRepository<AdministrativeDocument?, Long?>